src/main/java/roomescape/service/ThemeService.java

```diff
@@ -0,0 +1,78 @@
+package roomescape.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import roomescape.dto.ThemeRequestDto;
+import roomescape.dto.ThemeResponseDto;
+import roomescape.dto.ThemeUpdateRequestDto;
+import roomescape.dto.ThemesResponseDto;
+import roomescape.model.Theme;
+import roomescape.repository.ReservationRepository;
+import roomescape.repository.ThemeRepository;
+
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class ThemeService {
+    private final ThemeRepository themeRepository;
+    private final ReservationRepository reservationRepository;
+
+    public ThemeService(ThemeRepository themeRepository, ReservationRepository reservationRepository) {
+        this.themeRepository = themeRepository;
+        this.reservationRepository = reservationRepository;
+    }
+
+    @Transactional
+    public Long createTheme(ThemeRequestDto themeRequest) {
+        checkForSameNameTheme(themeRequest);
+        Theme theme = themeRequest.toEntity();
+        return themeRepository.save(theme);
+    }
+
+    public ThemeResponseDto findTheme(Long themeId) {
+        Theme theme = findThemeByIdOrThrowException(themeId);
+        return new ThemeResponseDto(theme);
+    }
+
+    public ThemesResponseDto findAllTheme() {
+        return new ThemesResponseDto(themeRepository.findAll().stream().map(ThemeResponseDto::new).collect(Collectors.toList()));
```

Comment:
하나의 라인에 하나의 .을 사용하면 가독성이 좋아집니다.
인틸리제이로 브레이킹포인트를 설정하고 디버깅할 때나 코드를 작성할 때 오른쪽에 타입을 알려주어서 읽기 편하게 되니 참고해주세요 :)

```suggestion
        List<ThemeResponseDto> themeResponseDtos = themeRepository.findAll()
                .stream()
                .map(ThemeResponseDto::new)
                .collect(Collectors.toList());
        return new ThemesResponseDto(themeResponseDtos);
```